extension/js/common/composer.ts

```diff
@@ -545,18 +545,27 @@ export class Composer {
     });
   }
 
-  private encryptMsgAsOfDateIfSomeAreExpired = async (armoredPubkeys: string[]): Promise<Date | undefined> => {
+  private encryptMsgAsOfDateIfSomeAreExpired = async (acctEmail: string, armoredPubkeys: PubkeyResult[]): Promise<Date | undefined> => {
     // todo - disallow in certain situations
     const usableUntil: number[] = [];
     const usableFrom: number[] = [];
-    for (const armoredPubkey of armoredPubkeys) {
-      const { keys: [pub] } = await openpgp.key.readArmored(armoredPubkey);
+    const myKey = armoredPubkeys.find(a => a.email === acctEmail);
```

Comment:
Here I would go by `armoredPubkeys.find(ap => ap.isMine)` - else I feel it would end up too complex when we take into account our own email aliases, and allowing to use different pubkey depending on alias.. if all you're trying to figure out is if the key is user's or not, I think it would be simpler to be explicit.